R/error-checks.R

```diff
@@ -8,32 +8,48 @@ isSameLength <- function(...) {
 #' Check if the provided object is of certain type
 #'
 #' @param object An object or a list of objects
-#' @param type String  representation or Class of the type that should be checked for
+#' @param type String representation or Class of the type that should be checked for
+#' @param nullAllowed Boolean flag if \code{NULL} is accepted for the \code{object}. If \code{TRUE},
+#' \code{NULL} always returns \code{TRUE}, otherwise \code{NULL} returns \code{FALSE}. Default is \code{FALSE}
 #'
 #' @return TRUE if the object or all objects inside the list are of the given type.
 #' Only the first level of the given list is considered.
-isOfType <- function(object, type) {
+isOfType <- function(object, type, nullAllowed = FALSE) {

```

Comment:
I extended the `isOfType`-function by `nullAllowed` so NULLs within a list of objects can also be allowed.

Otherwise:
`isOfType(list("as", "dsads", NULL, "sd"), "character", nullAllowed = TRUE)` returns `FALSE`.